Product Management Service/Code/product-service/src/main/java/com/shopkeeper/productservice/service/ImageService.java

```diff
@@ -1,6 +1,90 @@
 package com.shopkeeper.productservice.service;
 
+import java.io.IOException;
+import java.util.Date;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import com.shopkeeper.productservice.dao.ImageRepository;
+import com.shopkeeper.productservice.dao.ProductRepository;
+import com.shopkeeper.productservice.dto.Product;
+import com.shopkeeper.productservice.dto.ProductImages;
 
 @Service
-public class ImageService {}
+@Transactional
+public class ImageService {
+
+  @Autowired ImageRepository imageRepository;
+
+  @Autowired ProductRepository productRepository;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ImageService.class);
+
+  public ResponseEntity<List<ProductImages>> getImagesByProductId(Integer id) {
+    Product product = productRepository.findById(id).orElse(null);
+    if (product != null
+        && product.getProductImages() != null
+        && !product.getProductImages().isEmpty()) {
+      return new ResponseEntity<>(product.getProductImages(), HttpStatus.OK);
+    } else {
+      return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
```

Comment:
in an ideal situation, you should be having a product id in this table as a foreign key and pulling directly by that